### PR TITLE
[Enhancement] Cleanup load map style tasks

### PR DESCRIPTION
**Pull request by heshan0131**
- Add `requestMapStyles` action
- Consolidate style loading using `getLoadMapStyleTasks` called by `requestMapStyles` and `receiveMapConfg`
- Add tests

### CODE REVIEW

File: src/reducers/map-style-updaters.js [hunk 1]
```diff
@@ -171,17 +177,25 @@ function getMapStyles({
...
+  const buildingColor =
+    backgroundLayer &&
+    backgroundLayer.paint &&
+    backgroundLayer.paint['background-color']
+      ? backgroundLayer.paint['background-color']
+      : DEFAULT_BLDG_COLOR;
   // brighten or darken building based on style
-  const operation = style.id.match(/(?=(dark|night))/) ? 'brighter':  'darker';
+  const operation = style.id.match(/(?=(dark|night))/) ? 'brighter' : 'darker';
   const rgbObj = rgb(buildingColor)[operation]([0.2]);
```
Comment (macrigiuseppe) [inline src/reducers/map-style-updaters.js]: maybe it's better to create a constant for the 0.2 value

File: src/reducers/map-style-updaters.js [hunk 2]
```diff
@@ -289,6 +314,11 @@ export const loadMapStylesUpdater = (state, action) => {
 // do nothing for now, if didn't load, skip it
 export const loadMapStyleErrUpdater = (state, action) => state;
```
Comment (macrigiuseppe) [inline src/reducers/map-style-updaters.js]: delete __action__ if you are not using it

File: src/reducers/map-style-updaters.js [hunk 3]
```diff
@@ -368,19 +402,24 @@ export const resetMapConfigMapStyleUpdater = (state) => {
...
-      style: JSON.parse(JSON.stringify(style)),
-      label: style.name,
-      // gathering layer group info from style json
-      layerGroups: getLayerGroupsFromStyle(style)
-    } : {}),
+    ...(style
+      ? {
+          id: style.id || generateHashId(),
+          // make a copy of the style object
+          style: JSON.parse(JSON.stringify(style)),
```
Comment (macrigiuseppe) [inline src/reducers/map-style-updaters.js]: you are trying to clone an object here. Cloning directly a js object is more performant that using JSON.
https://jsperf.com/deep-copy-vs-json-stringify-json-parse/5

Lodash has several methods to support clone and deep clone

Comment (heshan0131) [reply src/reducers/map-style-updaters.js]: done

**Pull request merged**